io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -896,10 +896,33 @@ execAtomically !time !tid !tlbl !nextVid0 action0 k0 =
           go ctl' read written' writtenSeq' createdOuterSeq nextVid (k x)
 
       ThrowStm e ->
+        {-# SCC "execAtomically.go.ThrowStm" #-}
+        case ctl of 
+          AtomicallyFrame -> do
+            -- Revert all the TVar writes
+            !_ <- traverse_ (\(SomeTVar tvar) -> revertTVar tvar) written
```

Comment:
This is shared between all branches, maybe do it before the case branch?

Reply:
That's a good idea.

Reply:
Done for both IOSim/Internal.hs and IOSimPOR/Internal.hs